### DISCUSSION TITLE
Allow to globally set translateLabel

### DISCUSSION THREAD

**Discussion by jeffreyvr** [filamentphp/filament#10123] [Ideas]
Hi!

Perhaps this is already possible and I've simply overlooked it, but in general my idea is:

Allow `translateLabel` to be configurable globally. If you're making a multi language application, you pretty much want to run all labels through translation. 

As far as I'm aware, I need to call the `translateLabel` method every time specially. My suggestion, and I'm willing to contribute a PR for this, would be to:

- Add a config option to set `translateLabel` to true by default
- Add the logic to the `HasLabel` trait

**Comment by lcamero** [comment]
This would be great, it seems like the `HasLabel` trait doesn't expose any global way to set it to true really, only calling `translateLabel` every time

**Comment by jeffreyvr** [comment]
I've since figured out that you do this through `configureUsing`, for example, in your `AppServiceProvider` add this to `boot` function:

```php
Field::configureUsing(function(Field $field): void {
    $field->translateLabel();
});
```

**Reply by lcamero** [reply]
That's great thank you. I ended up doing the same for a few others too that I needed 👍 

```PHP
Column::configureUsing(function(Column $column): void {
    $column->translateLabel();
});
Filter::configureUsing(function(Filter $filter): void {
    $filter->translateLabel();
});
Field::configureUsing(function(Field $field): void {
    $field->translateLabel();
});
Entry::configureUsing(function(Entry $entry): void {
    $entry->translateLabel();
});
```

**Reply by faydanube** [reply]
thx!

**Reply by RoberRaf** [reply]
Thank you both, it made my life easier :D <3

**Reply by nwrman** [reply]
Thanks!!! I needed this two projects ago!

**Reply by JVillator0** [reply]
This is still working with filament v4? someone can confirm pls?
